packages/superset-ui-legacy-preset-chart-deckgl/src/CategoricalDeckGLContainer.jsx

```diff
@@ -179,7 +180,7 @@ export default class CategoricalDeckGLContainer extends React.PureComponent {
       data: { ...payload.data, features },
     };
 
-    return [getLayer(fd, filteredPayload, onAddFilter, this.setTooltip)];
+    return [getLayer(fd, filteredPayload, onAddFilter, this.setTooltip, this.props.datasource)];
```

Comment:
Why does this function return this.props.datasource? Can't we just pass it through props?